src/Analyzers/CSharp/Analyzers/RemoveUnnecessarySuppressions/CSharpRemoveUnnecessarySuppressionsDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.RemoveUnnecessarySuppressions;
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis.CSharp.RemoveUnnecessarySuppressions
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class CSharpRemoveUnnecessarySuppressionsDiagnosticAnalyzer
+        : AbstractRemoveUnnecessarySuppressionsDiagnosticAnalyzer
+    {
+        protected override void RegisterAttributeSyntaxAction(CompilationStartAnalysisContext context, CompilationAnalyzer compilationAnalyzer)
+        {
+            context.RegisterSyntaxNodeAction(context =>
```

Comment:
❔ Why not use an operation action for this?

Reply:
IOperation API is not yet supported for attributes - https://github.com/dotnet/roslyn/issues/18198. Tagging @333fred as well.